### ISSUE TITLE
CONTOSO functions not loading - #NAME? errors

### ISSUE THREAD

**Issue by adam-lerner51** [OfficeDev/office-js-docs-pr#258]
## Article URL

https://docs.microsoft.com/en-us/office/dev/add-ins/excel/custom-functions-overview

## Issue

 I've followed the instructions as outlined in the article above:
1.  Published the js, json, and html file onto localhost,
2.  Replaced the manifest file's URL references with my local host and set it up in a shared folder... triple checked the URLS are correct.
3.  When adding it via Excel (insert -> add-in -> shared folder -> add contoso the right sidebar pops up with the HTML text
4.  When trying to type in CONTOSO.ADD42 (1,1) I receive a #NAME? error - typing in CONT doesn't auto-suggest any CONTOSO functions

Any assistance would be greatly appreciated.

**Comment by TCourtneyOwen** [comment]
Hi Adam,

Did you perform this step from the ReadMe:

4. After Excel launches, you will need to register the custom-functions add-in to work around a bug:
    a. On the upper-left-hand side of Excel, there is a small hexagon icon with a dropdown arrow. The icon is to right of the Save icon.
    b. Click on this dropdown arrow and then click on the Custom Functions Sample add-in to register it.

I am betting this is the problem. Please let me know if this doesn't work for you.

Thanks,

Courtney

**Comment by adam-lerner51** [comment]
Hi Courtney,

Thanks for the prompt response.  It took some time to find the hexagonal shape (not appearing in my quick access area) but I realized I've already performed this step via the Insert tab.  Once I add/register the add-in  the HTML text "Thanks for trying custom function for Excel  You've installed!..." pops up in a window to the right within Excel but I still can't access the functions

**Comment by TCourtneyOwen** [comment]
HI Adam,

Based on the version of Excel you are running (1807), it sounds like you need to upgrade a newer version.  The following is listed as a prerequisite in  the ReadMe:

Install Office 2016 for Windows and join the [Office Insider](https://products.office.com/en-us/office-insider) program. You must have Office build number **8711 or later**.

If you want to attach your custom functions project folder in a zip file and send it to me on this thread, I can try it out on my end to confirm if this indeed a problem with your version  of Excel.

Thanks,

Courtney

**Comment by adam-lerner51** [comment]
Hi Courtney,

The Readme states "You must also have Office build build 9325+ on Windows" and I just updated to Build 10717.20011 (previously I had 10xxx - I think the original post was a typo).  I just joined the Office Insider "Insider" level so I should have the most up to date version.

Zip of the website and manifest is attached below.

Below is an image of my excel version... if I recall my IT team installed Office 365 on top of a static version of Excel 2016 - could that be an issue?

![image](https://user-images.githubusercontent.com/36620564/43153361-c4e74a52-8f79-11e8-9c64-349cb566d549.png)



[Example.zip](https://github.com/OfficeDev/office-js-docs-pr/files/2224555/Example.zip)

**Comment by adam-lerner51** [comment]
Happy days - I got it working!  Can't say I'm clear on what made the difference (only thing I can think of is I upgraded from the Monthly Stable level of the Insider Program to "Insider" level but it is working.

Thanks for your help Courtney!

![image](https://user-images.githubusercontent.com/36620564/43153763-dbc9cd7a-8f7a-11e8-803f-6ac308268b71.png)

**Comment by TCourtneyOwen** [comment]
Great, glad to hear it's working!  I am still going to take a look at your zip file to see if I can find anything glaringly wrong or missing. I will reply back a bit later with my findings.

Thanks,
Courtney

**Comment by TCourtneyOwen** [comment]
Hi Adam,

I was out of the office most of yesterday for a team event, so I am only getting a change to look at your project now. I am assuming you didn't include the node_modules directory in the zip file you sent me to decrease the size of the zip? If you are missing the node modules directory locally, the add-in won't work.

I will see what else might be causing custom-functions to fail in  your project and get back to you today.

Thanks,
Courtney

**Comment by TCourtneyOwen** [comment]
Hi Adam,

I am able to get your project working locally, but I too am having the same problem getting the async functions to work.  I am going to follow up and see if others are encountering this as well.

-Courtney

**Comment by adam-lerner51** [comment]
Hi Courtney,

First thanks for your help with all this - it really is appreciated and I hope you had fun on your team event!

I didn't include the aspnet_client folder and web config in the zip, but they are there, otherwise even the basic add42 wouldn't work I imagine.

It seems we reached the same point, where only aysnc functions aren't working.  I tried downgrading to the monthly stable build and nothing worked, then went back to the insider level Version 1808 (Build 10717.20011) and only the non-async function worked as before.

Which build were you having success with async functions?  Is there a way for me to force an update/downgrade to a certain build?

Also, the manifest file you sent over added folders to the URL references - FYI
![image](https://user-images.githubusercontent.com/36620564/43246400-379dc6de-90ba-11e8-8f65-8b599673e704.png)

Best,
Adam

**Comment by TCourtneyOwen** [comment]
Hi Adam,

I am using an internal build (10717.20003), which looks be a slightly older version than yours, and both the async and non-async methods work fine. The folders in the manifest file you cite shouldn't make a difference - it's just the default project structure when creating a custom functions project via [Yo Office](https://www.npmjs.com/package/generator-office).  The more I think of it, I am wondering if in fact you are creating your project via Yo Office? If not, can you please give that a try?  I fear that the repo at https://github.com/OfficeDev/Excel-Custom-Functions may be outdated, whereas I know the custom functions project installed via Yo Office is up to date.

-Courtney

**Comment by TCourtneyOwen** [comment]
I confirmed that the updated customfunctions.html on  https://github.com/OfficeDev/Excel-Custom-Functions does indeed fix the asycn functions issue.

-Courtney

**Comment by adam-lerner51** [comment]
Hi Courtney,

I tried using the new HTML file but I'm still not seeing success here unfortunately.  Would you be able to send over your JSON, JS, HTML, and Manifest file for me to look at?  Is there a way for me to add some kind of logger to the async process?  

Also, is there a way I can force update to a specific build (e.g. 10717.20003  as you mentioned above) via a command line?  

Best,
Adam

**Comment by TCourtneyOwen** [comment]
Hi Adam,

I am attaching my custom functions project that I created using the latest changes in the [custom functions repo]( https://github.com/OfficeDev/Excel-Custom-Functions).  The only changes I made to the to the files in the repo was updating the localhost url in customfunction.xml.  I don't think you will be able to get the exact version of Office I am using, since this build is only an internal build we use for testing. 
[Excel-Custom-Functions.zip](https://github.com/OfficeDev/office-js-docs-pr/files/2242230/Excel-Custom-Functions.zip)

I hope this zip file works for you.

-Courtney

**Comment by TCourtneyOwen** [comment]
That's great news, Adam!  Yes, that step about clearing the cache is not obvious and something we have on our radar to fix.  

If you want to get a version of the custom functions project that allows for loading in XL Online as well, I suggest installing Yo Office (npm install -g yo generator-office) and creating a custom functions project.  Information for Yo Office is available at https://github.com/OfficeDev/generator-office.  

In the very near future, we will be updating the custom functions project template on https://github.com/OfficeDev/Excel-Custom-Functions to match what Yo Office installs.  It's unfortunate that we currently have two instances of the custom functions template and that they aren't in sync.

Thanks,
Courtney

**Comment by AlexJerabek** [comment]
Hi @TCourtneyOwen,
Does your advice here also apply to issues https://github.com/OfficeDev/office-js-docs-pr/issues/244 and https://github.com/OfficeDev/office-js-docs-pr/issues/255 ?

**Comment by TCourtneyOwen** [comment]
Yes quite likely these issues are now fixed with the recent push of a new customfunctions.html to the repo

-Courtney
